js/src/data/reducer.js

```diff
@@ -627,6 +628,31 @@ const reducer = ( state = DEFAULT_STATE, action ) => {
 			);
 		}
 
+		case TYPES.RECEIVE_DETAILED_ERROR: {
+			const { errorSlot, error } = action;
+
+			return setIn( state, 'detailed_errors', [
+				...state.detailed_errors,
+
+				{
+					...error,
```

Comment:
@ankitrox I was thinking we should have an `error property which will have the error object/string as value.

```suggestion
					error,
```